edk2: 202402 -> 202408

mjoerg opened the pull request:
## Description of changes

edk2:
+ remove obsoleted patch
+ ensure patches to EDK II and OpenSSL are propagated to edk2.mkDerivation
+ update to 202405 with patch to avoid stuck system (#325403)
+ update to 202408

edk2-uefi-shell:
+ mark as broken on Darwin aarch64

OVMF:
+ fix build on Darwin x86-64
+ update Debian source used for building NVRAM variable templates with MSFT keys
+ expose NVRAM variable templates through passthru

## Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

<details>
  <summary>47 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
    <li>qemu_xen (qemu_xen_4_19)</li>
    <li>qemu_xen.debug (qemu_xen_4_19.debug)</li>
    <li>qemu_xen.ga (qemu_xen_4_19.ga)</li>
    <li>qemu_xen_4_16</li>
    <li>qemu_xen_4_16.debug</li>
    <li>qemu_xen_4_16.ga</li>
    <li>qemu_xen_4_17</li>
    <li>qemu_xen_4_17.debug</li>
    <li>qemu_xen_4_17.ga</li>
    <li>qemu_xen_4_18</li>
    <li>qemu_xen_4_18.debug</li>
    <li>qemu_xen_4_18.ga</li>
    <li>quickemu</li>
    <li>quickgui</li>
    <li>quickgui.debug</li>
    <li>quickgui.pubcache</li>
    <li>xen-guest-agent</li>
    <li>xen-slim</li>
    <li>xen-slim.boot</li>
    <li>xen-slim.dev</li>
    <li>xen-slim.doc</li>
    <li>xen-slim.man</li>
    <li>xenPackages.xen_4_16-slim</li>
    <li>xenPackages.xen_4_16-slim.boot</li>
    <li>xenPackages.xen_4_16-slim.dev</li>
    <li>xenPackages.xen_4_16-slim.doc</li>
    <li>xenPackages.xen_4_16-slim.man</li>
    <li>xenPackages.xen_4_17-slim</li>
    <li>xenPackages.xen_4_17-slim.boot</li>
    <li>xenPackages.xen_4_17-slim.dev</li>
    <li>xenPackages.xen_4_17-slim.doc</li>
    <li>xenPackages.xen_4_17-slim.man</li>
    <li>xenPackages.xen_4_18-slim</li>
    <li>xenPackages.xen_4_18-slim.boot</li>
    <li>xenPackages.xen_4_18-slim.dev</li>
    <li>xenPackages.xen_4_18-slim.doc</li>
    <li>xenPackages.xen_4_18-slim.man</li>
  </ul>
</details>

## Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>quickemu</li>
  </ul>
</details>

## Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2-uefi-shell</li>
    <li>quickemu</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>edk2</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

adamcstephens reviewed (changes requested):
There are too many changes in here to also include nixfmt. I understand it can be painful to work on unformatted files, but you should redo this without the formatting. If you really must format first, it would probably be easier to review in a standalone PR that also adds the commit to the `.git-blame-ignore-revs`

Given your interest and understanding of this area, you should really consider adding yourself to maintainers.

mjoerg committed:
edk2: 202402 -> 202405

https://github.com/tianocore/edk2/releases/tag/edk2-stable202405
https://github.com/tianocore/edk2/compare/edk2-stable202402...edk2-stable202405

Add patch to avoid stuck system (#325403).

mjoerg committed:
OVMFFull: fix build on Darwin

edk2-vars-generator might fail on Darwin due to passing an incompatible locale to mtools.
LC_ALL=C attempts to bypass this issue.

mjoerg committed:
OVMFFull: switch debian source to GitLab

The referenced archive is no longer available.

mjoerg committed:
edk2, edk2-uefi-shell, OVMF: add mjoerg as maintainer

mjoerg committed:
OVMFFull: update debian source to 2024.05-1

mjoerg committed:
edk2: remove patch obsoleted by OpenSSL 3.0.14

mjoerg committed:
OVMF: enable compilation using Clang

Disable -Wno-unneeded-internal-declaration causing the build to fail due to -Wall with Clang.

mjoerg committed:
edk2: propagate patches

Ensure patches to EDK II and OpenSSL are propagated to edk2.mkDerivation.

mjoerg committed:
edk2-uefi-shell: mark as broken on Darwin aarch64

mjoerg committed:
edk2: 202405 -> 202408

https://github.com/tianocore/edk2/releases/tag/edk2-stable202408
https://github.com/tianocore/edk2/compare/edk2-stable202408..edk2-stable202405

mjoerg committed:
OVMFFull: expose variablesMs similar to firmware and variables

LunNova commented:
Result of `nixpkgs-review pr 329338` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>47 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
    <li>qemu_xen (qemu_xen_4_19)</li>
    <li>qemu_xen.debug (qemu_xen_4_19.debug)</li>
    <li>qemu_xen.ga (qemu_xen_4_19.ga)</li>
    <li>qemu_xen_4_16</li>
    <li>qemu_xen_4_16.debug</li>
    <li>qemu_xen_4_16.ga</li>
    <li>qemu_xen_4_17</li>
    <li>qemu_xen_4_17.debug</li>
    <li>qemu_xen_4_17.ga</li>
    <li>qemu_xen_4_18</li>
    <li>qemu_xen_4_18.debug</li>
    <li>qemu_xen_4_18.ga</li>
    <li>quickemu</li>
    <li>quickgui</li>
    <li>quickgui.debug</li>
    <li>quickgui.pubcache</li>
    <li>xen-guest-agent</li>
    <li>xen-slim</li>
    <li>xen-slim.boot</li>
    <li>xen-slim.dev</li>
    <li>xen-slim.doc</li>
    <li>xen-slim.man</li>
    <li>xenPackages.xen_4_16-slim</li>
    <li>xenPackages.xen_4_16-slim.boot</li>
    <li>xenPackages.xen_4_16-slim.dev</li>
    <li>xenPackages.xen_4_16-slim.doc</li>
    <li>xenPackages.xen_4_16-slim.man</li>
    <li>xenPackages.xen_4_17-slim</li>
    <li>xenPackages.xen_4_17-slim.boot</li>
    <li>xenPackages.xen_4_17-slim.dev</li>
    <li>xenPackages.xen_4_17-slim.doc</li>
    <li>xenPackages.xen_4_17-slim.man</li>
    <li>xenPackages.xen_4_18-slim</li>
    <li>xenPackages.xen_4_18-slim.boot</li>
    <li>xenPackages.xen_4_18-slim.dev</li>
    <li>xenPackages.xen_4_18-slim.doc</li>
    <li>xenPackages.xen_4_18-slim.man</li>
  </ul>
</details>

LunNova commented:
Result of `nixpkgs-review pr 329338` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages marked as broken and skipped:</summary>
  <ul>
    <li>qemu_xen</li>
    <li>qemu_xen.ga</li>
    <li>qemu_xen_4_16</li>
    <li>qemu_xen_4_16.ga</li>
    <li>qemu_xen_4_17</li>
    <li>qemu_xen_4_17.ga</li>
    <li>qemu_xen_4_18</li>
    <li>qemu_xen_4_18.ga</li>
    <li>qemu_xen_4_19</li>
    <li>qemu_xen_4_19.ga</li>
    <li>xen-guest-agent</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
  </ul>
</details>

adamcstephens commented:
We should be safe from the channel blocker we broke before, now that it's tested.
